plugins/nodes/src/js/components/NodesTable.tsx

```diff
@@ -159,124 +166,135 @@ export default class NodesTable extends React.Component<
-            cellRenderer={typeRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">Region</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "region")}
-                sortDirection={sortColumn === "region" ? sortDirection : null}
-              />
-            }
-            cellRenderer={this.regionRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">Zone</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "zone")}
-                sortDirection={sortColumn === "zone" ? sortDirection : null}
-              />
-            }
-            cellRenderer={zoneRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">Tasks</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "tasks")}
-                sortDirection={sortColumn === "tasks" ? sortDirection : null}
-                textAlign="right"
-              />
-            }
-            cellRenderer={tasksRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">CPU</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "cpu")}
-                sortDirection={sortColumn === "cpu" ? sortDirection : null}
-              />
-            }
-            cellRenderer={cpuRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">Mem</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "mem")}
-                sortDirection={sortColumn === "mem" ? sortDirection : null}
-              />
-            }
-            cellRenderer={memRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">Disk</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "disk")}
-                sortDirection={sortColumn === "disk" ? sortDirection : null}
-              />
-            }
-            cellRenderer={diskRenderer}
-          />
-
-          <Column
-            header={
-              <SortableHeaderCell
-                columnContent={<Trans render="span">GPU</Trans>}
-                sortHandler={this.handleSortClick.bind(null, "gpu")}
-                sortDirection={sortColumn === "gpu" ? sortDirection : null}
-              />
-            }
-            cellRenderer={gpuRenderer}
-          />
-
-          <Column
-            header={<span title="Spacing" />}
-            cellRenderer={spacingRenderer}
-            width={spacingSizer}
-          />
-        </Table>
+        <NodesTableWrapper data={data.slice()}>
```

Comment:
I'm concerned about the tradeoff you mentioned in the description. This change seems to double the number of async requests to the nodes endpoint, but that seems like a sufficiently high cost to restructure things.

Reply:
There should be no redundant network requests. The request is shared via rxjs.

Reply:
I double checked. The polling of the nodes endpoint happen roughly twice that of master branch

Reply:
Okay, the problem I see here is that we are not requesting the `/net/v1/nodes` endpoint on the master branch, could it be that you are just filtering for `nodes` and not `/net/v1/nodes` since we also request `/system/health/v1/nodes` in that case the double number of request is a expected outcome.

Reply:
No, I'm seeing the normal request to /system/health/v1/nodes and two additional requests that happen nearly simultaneously to /net/v1/nodes

You can see here it's illustrated in the network waterfall and total request count after 80ish seconds

master branch:
![screen shot 2019-02-27 at 9 53 41 am](https://user-images.githubusercontent.com/174332/53507977-27cfea00-3a76-11e9-8e3e-9ce993b0bae6.png)

you branch:
![screen shot 2019-02-27 at 9 55 42 am](https://user-images.githubusercontent.com/174332/53507991-2d2d3480-3a76-11e9-8618-8e326198f167.png)

Reply:
Okay now I know why and how this happens. So we are requesting only once per update cycle. the unfortunate part is that we force the update 3 times in a cycle. :-( this has to do with how the data flows in the nodes page. The only way I know of to circle around this is to do a distinct until changed and check if the data has changed with `JSON.stringify` which is far to slow unfortunately. If you have another suggestion I am happy to address.

Reply:
I spent some time thinking about this and don't have a great answer. Have you looked at the componentWillReceiveProps method in NodesTableContainer could be short circuited if the mesos stream data is unchanged?